client/deviceCache.go

```diff
@@ -184,6 +182,27 @@ func (c *DeviceCache) GetDevicesFoundByIP() map[string]string {
 	return devices
 }
 
+func (c *DeviceCache) LoadAndDeleteDevices(deviceIDFilter []string) []*core.Device {
+	devices := make([]*core.Device, 0, len(deviceIDFilter))
+	if len(deviceIDFilter) == 0 {
+		for _, val := range c.devicesCache.PullOutAll() {
+			d := val.(*core.Device)
+			devices = append(devices, d)
+		}
+		return devices
+	}
+	for _, deviceID := range deviceIDFilter {
```

Comment:
The deviceIDFilter is a bit confusing name because under a filter I would imagine a functionality like wildcards `1234-*` and this is only a set of devices. Could we change the name to deviceIDs?